index.jsx

```diff
@@ -1,15 +1,27 @@
 import React, {PropTypes} from 'react';
 
-function radio(name, selectedValue, onChange) {
+function radio(kwargs) {
+
   return React.createClass({
     render: function() {
+      const name = kwargs.name;
+
+      // We only want to assign these attribute to the <input />
+      // elements if they were specified explicitly.
+      const optional = {};
+      if(kwargs.hasOwnProperty("selectedValue")) {
```

Comment:
Gotcha.  Actually, w/ the current code, the same restrictions to the props still apply.  The code only operates on the following props:

* name
* onChange
* selectedValue

Any other properties will simply be ignored.  That is, they will not be passed onto the underlying ```<input />``` element.

I think the main question is: what should happen when ```selectedValue={undefined}```?  Should React treat this as a Controlled Field because the developer intentionally set that value?  It sounds like you're saying that the answer is: "No.  React should treat this as an Uncontrolled Field when ```selectedValue={undefined}```"